src/rust/engine/process_execution/src/remote_cache_tests.rs

```diff
@@ -9,64 +8,108 @@ use std::time::Duration;
+  MultiPlatformProcess, Platform, Process, ProcessMetadata,
 };
 
-struct RoundtripResults {
-  uncached: Result<FallibleProcessResultWithPlatform, String>,
-  maybe_cached: Result<FallibleProcessResultWithPlatform, String>,
+/// A mock of the local runner used for better hermeticity of the tests.
+#[derive(Clone)]
+struct MockLocalCommandRunner {
+  result: Result<FallibleProcessResultWithPlatform, String>,
+  call_counter: Arc<Mutex<u32>>,
 }
 
-fn create_local_runner() -> (Box<dyn CommandRunnerTrait>, Store, TempDir, StubCAS) {
-  let runtime = task_executor::Executor::new();
-  let base_dir = TempDir::new().unwrap();
-  let named_cache_dir = base_dir.path().join("named_cache_dir");
-  let stub_cas = StubCAS::builder().build();
-  let store_dir = base_dir.path().join("store_dir");
-  let store = Store::with_remote(
-    runtime.clone(),
-    store_dir,
-    vec![stub_cas.address()],
-    None,
-    None,
-    None,
-    1,
-    10 * 1024 * 1024,
-    Duration::from_secs(1),
-    BackoffConfig::new(Duration::from_millis(10), 1.0, Duration::from_millis(10)).unwrap(),
-    1,
-    1,
-  )
-  .unwrap();
-  let runner = Box::new(crate::local::CommandRunner::new(
-    store.clone(),
-    runtime.clone(),
-    base_dir.path().to_owned(),
-    NamedCaches::new(named_cache_dir),
-    true,
-  ));
-  (runner, store, base_dir, stub_cas)
+impl MockLocalCommandRunner {
+  pub fn new(exit_code: i32, call_counter: Arc<Mutex<u32>>) -> MockLocalCommandRunner {
+    MockLocalCommandRunner {
+      result: Ok(FallibleProcessResultWithPlatform {
+        stdout_digest: EMPTY_DIGEST,
+        stderr_digest: EMPTY_DIGEST,
+        exit_code,
+        output_directory: EMPTY_DIGEST,
+        execution_attempts: vec![],
+        platform: Platform::current().unwrap(),
+      }),
+      call_counter,
+    }
+  }
+}
+
+#[async_trait]
+impl CommandRunnerTrait for MockLocalCommandRunner {
+  async fn run(
+    &self,
+    _req: MultiPlatformProcess,
+    _context: Context,
+  ) -> Result<FallibleProcessResultWithPlatform, String> {
+    let mut calls = self.call_counter.lock();
+    *calls += 1;
+    self.result.clone()
+  }
+
+  fn extract_compatible_request(&self, req: &MultiPlatformProcess) -> Option<Process> {
+    Some(req.0.get(&None).unwrap().clone())
+  }
+}
+
+// NB: We bundle these into a struct to ensure they share the same lifetime.
+struct StoreSetup {
+  pub store: Store,
+  pub store_dir: PathBuf,
+  pub cas: StubCAS,
+}
+
+impl StoreSetup {
+  pub fn new() -> StoreSetup {
+    let runtime = task_executor::Executor::new();
+    let cas = StubCAS::builder().build();
+    let store_dir = TempDir::new().unwrap().path().join("store_dir");
+    let store = Store::with_remote(
+      runtime,
+      store_dir.clone(),
+      vec![cas.address()],
+      None,
+      None,
+      None,
+      1,
+      10 * 1024 * 1024,
+      Duration::from_secs(1),
+      BackoffConfig::new(Duration::from_millis(10), 1.0, Duration::from_millis(10)).unwrap(),
```

Comment:
As in, share the constants across files? Or factor these up into constants? Imo the constants wouldn't be worth it very much because they're not used elsewhere.

Reply:
Share across files if you think it is worth it? My thought was DRY for test constants.

Reply:
Indeed, it looks like the code to create a remote store is very similar across the two files. I think it could be worth factoring up, but probably as a second PR. This diff is already bad enough.